appgate/syncer/git.py

```diff
@@ -134,32 +141,60 @@ def create_or_update_pull_request(
         pull_request: PullRequest | None,
         commits: Dict[str, List[Tuple[str, GitCommitState]]],
     ) -> PullRequest | None:
-        pass
+        raise NotImplementedError()
 
 
-def github_repo(ctx: GitOperatorContext, repository_path: Path) -> GitRepo:
-    token = ensure_env(GITHUB_TOKEN_ENV)
+def clone_repo(
+    ctx: GitOperatorContext, vendor: GitVendor, repository_path: Path
+) -> Repo:
     repository = ctx.git_repository_fork or ctx.git_repository
     log.info("[git-operator] Initializing the git repository by cloning %s", repository)
     if repository_path.exists():
         shutil.rmtree(repository_path)
-    if not GITHUB_DEPLOYMENT_KEY_PATH.exists():
-        raise AppgateException(
-            f"Unable to find deployment key {GITHUB_DEPLOYMENT_KEY_PATH}"
-        )
+    if not GIT_SSH_KEY_PATH.exists():
+        raise AppgateException(f"Unable to find SSH key {GIT_SSH_KEY_PATH}")
     try:
         git_repo = Repo.clone_from(
-            f"git@github.com:{repository}",
+            f"git@{vendor}.com:{repository}",
```

Comment:
This works now but maybe would make sense to have a mapping of git vendor to URL instead? or a function.
Not a big deal now but this will break if using something that doesn't end in .com.

Reply:
:+1:

We can fix that in future prs